server/service/live_queries.go

```diff
@@ -55,14 +56,34 @@ func runLiveQueryEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 		},
 	}
 
-	queryResults, respondedHostCount := svc.RunLiveQueryDeadline(ctx, req.QueryIDs, req.HostIDs, duration)
+	queryResults, respondedHostCount, err := svc.RunLiveQueryDeadline(ctx, req.QueryIDs, req.HostIDs, duration)
+	if err != nil {
+		return nil, err
+	}
+	// Check if all query results were forbidden due to lack of authorization
+	allResultsForbidden := len(queryResults) > 0
+	for _, r := range queryResults {
+		if r.Error == nil || *r.Error != authz.ForbiddenErrorMessage {
+			allResultsForbidden = false
+			break
+		}
+	}
+	if allResultsForbidden {
```

Comment:
Could we add an integration test for this scenario? in `server/service/integration_live_queries_test.go`

Reply:
Test added.